Apps/CocoaPods-FCM/src/View/SettingsView.swift

```diff
@@ -50,57 +52,52 @@ struct SettingsView: View {
                 ColorButton("Save") {
                     hideKeyboard() // makes all textfields lose focus so that @State variables are up-to-date with the textfield values.
 
-                    guard viewModel.settings.bqSecondsDelay > 0 else {
+                    guard viewModel.settings.flushInterval > 0 else {
                         alertMessage = "BQ seconds delay must be > 0"
                         return
                     }
 
-                    guard viewModel.settings.bqMinNumberTasks > 0 else {
+                    guard viewModel.settings.flushAt > 0 else {
                         alertMessage = "BQ min number tasks must be > 0"
                         return
                     }
 
-                    guard verifyTrackUrl() else {
+                    guard verifyHost(isCDN: true) else {
                         return
                     }
 
+                    nonBlockingMessage = "Settings saved. This will require an app restart to bring the changes in effect."
                     viewModel.saveSettings()
 
-                    // Re-initialize the SDK to make the config changes go into place immediately
-                    CustomerIO.initialize(siteId: viewModel.settings.siteId, apiKey: viewModel.settings.apiKey, region: .US) { config in
-                        viewModel.settings.configureCioSdk(config: &config)
-                    }
-
                     let didChangeSiteId = siteIdBeforeEditingSettings != viewModel.settings.siteId
                     if didChangeSiteId { // if siteid changed, we need to re-identify for the Customer.io SDK to get into a working state.
                         userManager.logout()
                     }
-
-                    done()
+                    timer.start(interval: TimeInterval(3)) {
+                        done()
+                    }
                 }.setAppiumId("Save Settings Button")
 
                 Button("Restore default settings") {
                     viewModel.restoreDefaultSettings()
                 }.setAppiumId("Restore Default Settings Button")
             }
+
             .padding([.leading, .trailing], 10)
             .onAppear {
-                siteIdBeforeEditingSettings = CustomerIO.shared.siteId!
+                siteIdBeforeEditingSettings = BuildEnvironment.CustomerIO.siteId
```

Comment:
i don't think it would work for the second, or third time you change siteId ? because it would always check for the first one?

Reply:
I am not very clear about the question, but let  me try answering to what I understood. Feel free to correct me. 

This is to restore default value of `siteId` and not to fetch the current value of `siteId`. The default value is the one that we capture in BuildEnvironment.